collect.py

```diff
@@ -112,13 +113,23 @@ def do_something(driver, elem_attributes=None):
         for child in children:
             # Get all the attributes of the child.
             elem_attributes = get_all_attributes(driver, child)
+            path_to_follow.append(elem_attributes)
+
+            # We check the nodes if previously visited or not
+            if frozenset([frozenset(attributes) for attributes in path_to_follow]) in visited_path:
```

Comment:
I think this might be imprecise. What if there are two buttons with the same attributes on two different pages? We'll consider them as being the same button.
I would remove the visited checks until we can think of a more precise solution.

There's also a problem with the current code. `attributes` is an object where the keys are the names of the attributes and the values are the values of the attributes. frozenset({'key': 'value'}) is frozenset({'key'}). We should consider both the names and the values to identify elements and not only the names.

The code is also not very readable. I suggest moving the visited check and the visited addition into separate functions (something like `was_visited` and `visit`).

Reply:
Oh, I see. You were not trying to implement a way to check whether we already visited a page, but whether we already followed a path.
OK, makes more sense then. The latest two paragraphs still apply.

Reply:
There was another comment from me here, which somehow got removed.
In the comment, I was saying I thought you were trying to do something else and so you could discard the first paragraph of my first comment.